services/wfm/src/main/java/org/openkilda/wfm/topology/event/service/IslService.java

```
@@ -0,0 +1,236 @@
+import org.openkilda.persistence.repositories.SwitchRepository;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Clock;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+
+public class IslService {
+
+    private static final Logger logger = LoggerFactory.getLogger(IslService.class);
+
+    private TransactionManager transactionManager;
+    private IslRepository islRepository;
+    private LinkPropsRepository linkPropsRepository;
+    private SwitchRepository switchRepository;
+
+    public IslService(TransactionManager transactionManager, RepositoryFactory repositoryFactory) {
+        this.transactionManager = transactionManager;
+        this.islRepository = repositoryFactory.createIslRepository();
+        this.linkPropsRepository = repositoryFactory.createLinkPropsRepository();
+        this.switchRepository = repositoryFactory.createSwitchRepository();
+    }
+
+    /**
+     * Create or update isl.
+     *
+     * @param isl isl.
+     * @return transaction was successful or not.
+     */
+    public boolean createOrUpdateIsl(Isl isl) {
+        logger.info("Create or update ISL {}_{}-{}_{}",
+                isl.getSrcSwitchId(), isl.getSrcPort(), isl.getDestSwitchId(), isl.getDestPort());
+        transactionManager.begin();
+        try {
+            processCreateOrUpdateIsl(isl);
+            transactionManager.commit();
+        } catch (Exception e) {
+            transactionManager.rollback();
+            return false;
+        }
+
+        transactionManager.begin();
+        try {
+            islRepository.setLinkPropsToIsl(isl);
+            islRepository.updateIslBandwidth(isl.getSrcSwitchId(), isl.getSrcPort(),
+                    isl.getDestSwitchId(), isl.getDestPort());
+            transactionManager.commit();
+        } catch (Exception e) {
+            transactionManager.rollback();
+            return false;
```

An exception must not be swallowed without a reason.

---

Fixed